packages/next/src/app-dir/server.ts

```diff
@@ -64,6 +67,15 @@ export function experimental_createTRPCNextAppDirServer<
   >;
 }
 
+/**
+ * Rethrow errors that should be handled by Next.js
+ */
+const rethrowNextErrors = (error: TRPCError) => {
+  const { cause } = error;
+  if (isRedirectError(cause) || isNotFoundError(cause)) {
+    throw error.cause;
+  }
+};
```

Comment:
cc @juliusmarminge you probably want some variant of this if you're throwing errors in RSC